src/pocketmine/entity/Human.php

```diff
@@ -501,6 +516,16 @@ public function saveNBT(){
 			$this->namedtag->SelectedInventorySlot = new IntTag("SelectedInventorySlot", $this->inventory->getHeldItemIndex());
 		}
 
+		if($this->enderChestInventory !== null){
+			$slotCount = $this->enderChestInventory->getSize();
+			for($slot = 0; $slot < $slotCount; ++$slot){
+				$item = $this->enderChestInventory->getItem($slot);
+				if($item->getId() !== ItemItem::AIR){
+					$this->namedtag->EnderChestInventory[$slot] = $item->nbtSerialize($slot);
```

Comment:
If the player's data was previously generated by PocketMine-MP (not spoons) this tag won't exist. This means that you're performing array-access on null, which results in the items silently being destroyed on quit.